docs/reference/module-overview.md

```diff
@@ -0,0 +1,284 @@
+# Anatomy of a module
+
+Module configuration objects may use the following configuration properties.
+
+## Configuration settings
+
+| Setting name | Value type | [Cascades](#cascading-settings) | Description | Module types |
+| ------- | ------- | ------- | ------- | :-----: |
+| [`extend`](#extend) | String | No | Name a base class module | All |
+| [`improve`](#improve) | String | No | Name a module to update its functionality | All |
+| [`options`](#options) | Object | No | Configure module options | All |
+| [`fields`](#fields) | Object | Yes | Configure doc type fields | Doc, Widget |
+| [`filters`](#filters) | Object | Yes | Configure piece type filters | Piece |
+| [`columns`](#columns) | Object | Yes | Configure piece type manager columns | Piece |
+| [`instantiate`](#instantiate) | Boolean | No | Prevent a module from being fully instantiated | All |
+
+### `extend`
+
+Name a module to use as a base class for your custom module. "Extending" a module this way makes all of the functionality and configuration of the base class available in the new module, other than its `name` property.
+
+The most common base class modules include:
+
+- `@apostrophecms/piece-type` to add a new custom piece type
```

Comment:
"Custom" here means that you are building it from scratch, rather than configuring a core or installed module. I agree it's not the best term for both parts of this sentence, surely.